lineapy/graph_reader/session_artifacts.py

```diff
@@ -34,18 +34,26 @@ class SessionArtifacts:
     nx_graph: nx.DiGraph
     artifact_nodecollections: List[NodeCollection]
     import_nodecollection: NodeCollection
-    artifact_list: List[LineaArtifact]
+    artifact_list: List[ArtifactORM]
     node_context: Dict[LineaID, NodeInfo]
     input_parameters: List[str]
     input_parameters_node: Dict[str, LineaID]
 
     def __init__(
-        self, artifacts: List[LineaArtifact], input_parameters: List[str] = []
+        self,
+        db: RelationalLineaDB,
+        artifacts: List[ArtifactORM],
+        input_parameters: List[str] = [],
     ) -> None:
+        self.db = db
         self.artifact_list = artifacts
-        self.session_id = artifacts[0]._session_id
-        self.db = artifacts[0].db
-        self.graph = artifacts[0]._get_graph()
+        self.session_id = artifacts[0].node.session_id
+
+        # Construct session graph
+        session_context = self.db.get_session_context(self.session_id)
+        nodes = self.db.get_nodes_for_session(self.session_id)
+        self.graph = Graph(nodes, session_context)
```

Comment:
is this the safer / more canonical way to get the Session Graph? Just wondering if this change is necessary (because self.graph is different after this change) or just better style ...

Reply:
This is because we now use `ArtifactORM` (instead of `LineaArtifact`) in `SessionArtifacts`, and `ArtifactORM` does not have `_get_graph()` method. So, I just adapted code from `LineaArtifact._get_graph()`, which should return the same graph object as before.

@lionsardesai This type of code adaptation/repetition is a consequence of using `ArtifactORM` instead of `LineaArtifact`. Are we good with it?

Reply:
lets refactor the _get_graph method here. its only used in 4 places and doesnt seem to use any private vars. Also its a private method and using it outside like this is an orange flag at the very least.

Here's my proposal: pull out _get_graph() method from LineaArtifact and put it inside the Graph class as a staticmethod to create a graph object from session (see @staticmethod in pythons)

Reply:
Refactoring done.